salt/modules/state.py

```diff
@@ -2286,7 +2288,85 @@ def _disabled(funs):
     return ret
 
 
-def event(tagmatch='*',
+@with_deprecated(globals(), "Fluorine")
+def event(tagmatch='.*',
+          count=-1,
+          quiet=False,
+          sock_dir=None,
+          pretty=False,
+          node='minion'):
+    r'''
+    Watch Salt's event bus and block until the given tag is matched
+
+    .. versionadded:: 2016.3.0
+
+    This is useful for utilizing Salt's event bus from shell scripts or for
+    taking simple actions directly from the CLI.
+
+    Enable debug logging to see ignored events.
+
+    :param tagmatch: the event is written to stdout for each tag that matches
+        this regular expression.
+    :param count: this number is decremented for each event that matches the
+        ``tagmatch`` parameter; pass ``-1`` to listen forever.
+    :param quiet: do not print to stdout; just block
+    :param sock_dir: path to the Salt master's event socket file.
+    :param pretty: Output the JSON all on a single line if ``False`` (useful
+        for shell tools); pretty-print the JSON output if ``True``.
+    :param node: Watch the minion-side or master-side event bus.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-call --local state.event pretty=True
+    '''
+    try:
+        patt = re.compile(tagmatch)
+    except:
+        raise SaltInvocationError(
+            'Invalid regular expression "{0}" given as parameter "tagmatch"'.format(tagmatch)
+        )
+
+    sevent = salt.utils.event.get_event(
+            node,
+            sock_dir or __opts__['sock_dir'],
+            __opts__['transport'],
+            opts=__opts__,
+            listen=True)
+
+    we_are_counting = True if count > -1 else False
+
+    while True:
+        ret = sevent.get_event(full=True, auto_reconnect=True)
+        if ret is None:
+            continue
+
+        if patt.search(ret['tag']) is not None:
+            if not quiet:
+                salt.utils.stringutils.print_cli(
+                    str('{0}\t{1}').format(  # future lint: blacklisted-function
+                        salt.utils.stringutils.to_str(ret['tag']),
+                        salt.utils.json.dumps(
+                            ret['data'],
+                            sort_keys=pretty,
+                            indent=None if not pretty else 4)
+                    )
+                )
+                sys.stdout.flush()
+
+            if we_are_counting:
+                count -= 1
+                if count == 0:
```

Comment:
First, you can just:
```python
if not count:
    break
```
But this is still wrong. The `-1` is "listen forever". Can be `count=0`? If yes, what does it mean then other, than "do not count"? Still, you do not need this `we_are_watching_some_other_variable`. Just this:

```python
if count > 0:
    count -= 1
if not count:
    break
```